# Football matches analysis

In [2]:
import numpy as np
import pandas as pd

In [3]:
df_home = pd.read_csv('../data/players_home_matches.csv')
df_away = pd.read_csv('../data/players_away_matches.csv')

In [15]:
col1 = 'team_name'
col2 = 'homeTeamName'

diff_rows = df_home.loc[df_home[col1] != df_home[col2]]
print(diff_rows)

Empty DataFrame
Columns: [Unnamed: 0, awayScore, awayTeamName, awayTeamTacticalSchema, homeScore, homeTeamName, homeTeamTacticalSchema, matchDate, matchWeek, playerName, aerials_lost, aerials_won, aerials_won_pct, age, assisted_shots, assists, ball_recoveries, blocked_passes, blocked_shots, blocked_shots_saves, blocks, cards_red, cards_yellow, cards_yellow_red, carries, carries_into_final_third, carries_into_penalty_area, carry_distance, carry_progressive_distance, clearances, corner_kicks, corner_kicks_in, corner_kicks_out, corner_kicks_straight, crosses, crosses_into_penalty_area, dispossessed, dribble_tackles, dribble_tackles_pct, dribbled_past, dribbles, dribbles_completed, dribbles_completed_pct, dribbles_vs, errors, fouled, fouls, gca, goals, interceptions, minutes, miscontrols, npxg, nutmegs, offsides, own_goals, pass_targets, passes, passes_blocked, passes_completed, passes_completed_long, passes_completed_medium, passes_completed_short, passes_dead, passes_free_kicks, passes_g

In [22]:
cols_to_move = ['championship_name', 'matchWeek', 'matchDate', 'homeTeamName', 'awayTeamName',
                'homeScore', 'awayScore', 'homeTeamTacticalSchema', 'awayTeamTacticalSchema', 'team_name', 'playerName']
cols_to_delete = ['Unnamed: 0']

In [23]:
new_col_order = cols_to_move + [col for col in df_home.columns if col not in cols_to_move + cols_to_delete]
df_home = df_home[new_col_order]
df_away = df_away[new_col_order]

In [37]:
df_away

championship_name  matchWeek   matchDate          homeTeamName  \
0               ligue 1          1  2021-08-08                  Nice   
1               ligue 1          1  2021-08-08                  Nice   
2               ligue 1          1  2021-08-08                  Nice   
3               ligue 1          1  2021-08-08                  Nice   
4               ligue 1          1  2021-08-08                  Nice   
...                 ...        ...         ...                   ...   
76180    premier league         11  2020-12-06  West Bromwich Albion   
76181    premier league         29  2021-03-04  West Bromwich Albion   
76182    premier league         37  2021-05-19  West Bromwich Albion   
76183    premier league         37  2021-05-19  West Bromwich Albion   
76184    premier league          3  2020-09-26  West Bromwich Albion   

          awayTeamName  homeScore  awayScore homeTeamTacticalSchema  \
0                Reims          0          0                (4-4-2)   
1                Reims          0          0                (4-4-2)   
2                Reims          0          0                (4-4-2)   
3                Reims          0          0                (4-4-2)   
4                Reims          0          0                (4-4-2)   
...                ...        ...        ...                    ...   
76180   Crystal Palace          1          5              (3-4-1-2)   
76181          Everton          0          1              (4-1-4-1)   
76182  West Ham United          1          3              (4-4-1-1)   
76183  West Ham United          1          3              (4-4-1-1)   
76184          Chelsea          3          3                (3-4-3)   

      awayTeamTacticalSchema             team_name  ... throw_ins  touches  \
0                  (3-4-1-2)                  Nice  ...       0.0     58.0   
1                  (3-4-1-2)                  Nice  ...      14.0     61.0   
2                  (3-4-1-2)                  Nice  ...       0.0     19.0   
3                  (3-4-1-2)                  Nice  ...       0.0     28.0   
4                  (3-4-1-2)                  Nice  ...       0.0     27.0   
...                      ...                   ...  ...       ...      ...   
76180                (4-4-2)  West Bromwich Albion  ...       0.0     32.0   
76181              (3-4-1-2)  West Bromwich Albion  ...       0.0     66.0   
76182              (4-2-3-1)  West Bromwich Albion  ...       1.0     34.0   
76183              (4-2-3-1)  West Bromwich Albion  ...       0.0      2.0   
76184              (4-2-3-1)  West Bromwich Albion  ...       0.0     59.0   

       touches_att_3rd  touches_att_pen_area  touches_def_3rd  \
0                  3.0                   1.0             37.0   
1                 19.0                   0.0             21.0   
2                  8.0                   2.0              4.0   
3                  2.0                   0.0              8.0   
4                  7.0                   1.0              9.0   
...                ...                   ...              ...   
76180             22.0                   5.0              2.0   
76181             25.0                   2.0              6.0   
76182              1.0                   1.0             25.0   
76183              0.0                   0.0              1.0   
76184             37.0                  10.0              3.0   

       touches_def_pen_area  touches_live_ball  touches_mid_3rd   xa   xg  
0                       8.0               57.0             27.0  0.0  0.0  
1                       2.0               47.0             28.0  0.0  0.0  
2                       1.0               19.0              7.0  0.1  0.0  
3                       3.0               28.0             18.0  0.0  0.0  
4                       1.0               22.0             11.0  0.0  0.0  
...                     ...                ...              ...  ...  ...  
76180                   2.0           

In [34]:
df_home.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 76215 entries, 0 to 76214
Data columns (total 126 columns):
 #    Column                       Dtype  
---   ------                       -----  
 0    championship_name            object 
 1    matchWeek                    int64  
 2    matchDate                    object 
 3    homeTeamName                 object 
 4    awayTeamName                 object 
 5    homeScore                    int64  
 6    awayScore                    int64  
 7    homeTeamTacticalSchema       object 
 8    awayTeamTacticalSchema       object 
 9    team_name                    object 
 10   playerName                   object 
 11   aerials_lost                 float64
 12   aerials_won                  float64
 13   aerials_won_pct              float64
 14   age                          float64
 15   assisted_shots               float64
 16   assists                      int64  
 17   ball_recoveries              float64
 18   blocked_passes          

### Our dataset consists of 126 columns where 128 are numeric and 8 are not 

### Not numeric columns are:
- championship_name 
- matchDate 
- homeTeamName                  
- awayTeamName 
- homeTeamTacticalSchema        
- awayTeamTacticalSchema        
- team_name                     
- playerName                    

### The rest of columns which are numeric mostly describe various match statistics for each individual player from the home and away team

### We decided to group these columns into a couple of categories:
- Offensive statistics
    - goals
    - assists
    - shots_total
    - shots_on_target
    - xg (oczekiwane gole)
    - xa (oczekiwane asysty)
    - gca (wielokrotne zdarzenia przyczyniające się do celnej akcji)
    - sca (wielokrotne zdarzenia przyczyniające się do strzału)
    - .................................
    - assisted_shots
    - carries
    - carries_into_final_third
    - carries_into_penalty_area
    - carry_distance
    - carry_progressive_distance
    - progressive_carries
    - dispossessed
    - dribble_tackles
    - dribble_tackles_pct
    - dribbled_past
    - errors
    - miscontrols
    - own_goals
    - pens_att
    - pens_conceded
    - pens_made
    - pens_won
    - players_dribbled_past
    - through_balls
    - corner_kicks
    - corner_kicks_in
    - corner_kicks_out
    - corner_kicks_straight
    - throw_ins

- Defensive statistics
    - interceptions
    - tackles
    - tackles_won
    - pressure_regains
    - pressures
    - ball_recoveries
    - blocks
    - blocked_shots
    - blocked_passes
    - clearances
    - .................
    - tackles_att_3rd
    - tackles_def_3rd
    - tackles_interceptions
    - tackles_mid_3rd
    - touches_att_3rd
    - touches_att_pen_area
    - touches_def_3rd
    - touches_def_pen_area
    - touches_live_ball
    - touches_mid_3rd
    - pressures_att_3rd
    - pressures_def_3rd
    - pressures_mid_3rd
- Passes statistics
    - passes
    - passes_completed
    - passes_pct (procentowa skuteczność podań)
    - passes_progressive_distance (odległość postępujących podań)
    - passes_received (odebrane podania)
    - passes_received_pct (procentowa skuteczność odebranych podań)
    - passes_into_final_third
    - passes_into_penalty_area
    - progressive_passes (postępujące podania)
    - progressive_passes_received (odebrane postępujące podania)
    - ...........................................
    - passes_blocked
    - passes_completed_long
    - passes_completed_medium
    - passes_completed_short
    - passes_dead
    - passes_free_kicks
    - passes_ground
    - passes_head
    - passes_high
    - passes_intercepted
    - passes_left_foot
    - passes_live
    - passes_long
    - passes_low
    - passes_medium
    - passes_offsides
    - passes_oob
    - passes_other_body
    - passes_pct_long
    - passes_pct_medium
    - passes_pct_short
    - passes_pressure
    - passes_switches
    - passes_total_distance
- Dribble statistics
    - dribbles
    - dribbles_completed
    - dribbles_completed_pct
    - dribbles_vs
- Physical stats
    - aerials_lost
    - aerials_won
    - aerials_won_pct
    - fouled
    - fouls
    - nutmegs
    - offsides
- Other
    - age
    - minutes
    - matchWeek
    - homeScore
    - awayScore